driver/stub/stub_devconf.c

```diff
@@ -56,18 +54,15 @@ dup_info_intf(PUSBD_INTERFACE_INFORMATION info_intf)
 static BOOLEAN
 build_infos_intf(devconf_t *devconf, PUSBD_INTERFACE_INFORMATION infos_intf)
 {
-	PUSBD_INTERFACE_INFORMATION	info_intf;
 	unsigned	i;
 
-	info_intf = infos_intf;
 	for (i = 0; i < devconf->bNumInterfaces; i++) {
-		PUSBD_INTERFACE_INFORMATION	info_intf_copied = dup_info_intf(info_intf);
+		PUSBD_INTERFACE_INFORMATION	info_intf_copied = dup_info_intf(&infos_intf[i]);
```

Comment:
It's possible I don't have it right because the usb device doesn't work reliably although I wasn't sure if it was another issue. The device does enumerate fine in Linux now.

With the old code Windows crashes with "page fault in nonpaged area" on the second pass through the for loop when dup_info_intf()->INFO_INTF_SIZE()->(info_intf)->NumberOfPipes is accessed. 

I'll look again carefully.

Reply:
I used this code as the reference when looking at the usbip code. https://docs.microsoft.com/en-us/windows-hardware/drivers/usbcon/how-to-select-a-configuration-for-a-usb-device

Microsoft seems to use an array reference as well. 

`Interface = interfaceList[interfaceIndex].Interface;`

Reply:
My idea is wrong but I don't think either of us have it right.

Reply:
Microsoft states:

> USBD_SelectConfigUrbAllocateAndBuild performs the following tasks: 
> 1. Creates an URB and fills it with information about the specified configuration, its interfaces and endpoints, and sets the request type to URB_FUNCTION_SELECT_CONFIGURATION.
> 2. Fills a USBD_INTERFACE_INFORMATION structure in the URB for each interface.
> 3. Sets the Interface member of the nth element of the caller-provided USBD_INTERFACE_LIST_ENTRY array to the address of the corresponding USBD_INTERFACE_INFORMATION structure in the URB.

I really don't know what #2 means. _URB_SELECT_CONFIGURATION provides a single instance of USBD_INTERFACE_INFORMATION Interface. It's not clear if this is actually an array. 

What is made clear by the example source code and #3 is the information is also present in  PUSBD_INTERFACE_LIST_ENTRY pintf_list so that may be the safe location to get this information from.

Reply:
I added a second commit reflecting this change. The device seems much more reliable now.

Reply:
Yes, your modified code is right.
I had tested and confirmed that pintf.Interface points to a payload of an allocated urb after USBD_SelectConfigUrbAllocateAndBuild() call. Using pintf.Interface would be better than the old approach.

Reply:
I don't know why the old approach fails. Anyway, using pintf_list seems to be more elegant approach. Thanks..